### ISSUE TITLE
[14963] Add ability to notify multiple users when updating a ticket

### ISSUE THREAD

**Issue by Aycorn** [DeskDirector/UserVoice#281]
Have the ability for users to essentially add multiple users to tickets in order for all users to be included in all notifications and communications between the ticket owner and client contact.

**Comment by D-Irizarry** [comment]
This is a great idea and is especially important when trying to communicate between multiple affected users, or a user, their vendor and the MSP for example. Sometimes supervisors ask to be kept in the loop (internally or at the client side).

in Connectwise, you're able to add email addresses to a ticket and check / uncheck the box to include them on future updates, which is easy to modify on the fly and "sticks" so you don't have to add them back every time.

**Comment by Lowell-Syncrety** [comment]
We absolutely need this to be fully able to use DD Tech. Otherwise we're having to switch back to Staff HUD or CWM client if we just want to add another user to communicate to, or to CC an existing second user on a ticket.

**Comment by dr3dgen** [comment]
I'm seeing this request come up in a number of places here. We've stressed the idea of no more email ticket communication to our clients, but it's unfortunate we can't deliver that in part because of the workflow for tickets that involve multiple people. We end up defaulting to the CW system, and the CC'd end users can't see the ticket in their Portal. I think this is a HUGE missed opportunity.

**Comment by Andrew-Lahikainen** [comment]
@dr3dgen @Lowell-Syncrety I like this idea, we just have to work out the feasibility.

I like the idea of allowing Techs to assign additional contacts to tickets instead of simply implementing email CCs. Just some thoughts on this:

* I believe there is a limitation with the ConnectWise API which prevents us adding a note to a ticket from a contact that doesn't belong to the same company as the assigned contact. This will mean that external CCs will have a readonly version if they choose to view it in the Portal.

* Company boundaries are simple to understand and implement. As soon as this rule is broken, complexity increases. We've got tentative plans to implement groups, and we're definitely seeing a lot of complexity in how companies/groups/sites interact with regards to permissions and search etc. Even UX and support suffers here as people are going to have a harder time understanding the system and why certain contacts can see certain tickets.

@Nness do you have any thoughts on this feature? I think it wouldn't be to hard to implement CCs as ConnectWise does, but what do you think on allowing CCs to view the tickets in Portal?

**Comment by dr3dgen** [comment]
@Andrew-Lahikainen For my purposes, it is exclusively internal contacts within a company. Coordinating with officer managers and an employee; configuring new hire IT equipment between HR, IT, and department management; shutting down accounts during employee exits; ordering new equipment; and communicating on change requests with key people in a specific departments. These are a few of the scenarios we typically encounter. (I'm realizing as I type this that we probably need to implement approvals.)

tl;dr Internal use only per company is fine for us.

Honestly, I'd be fine with only allowing CCs on the Tech side after the initial ticket comes in. Then the resolution or progress can be communicated to those impacted. @Lowell-Syncrety would something like that work for your team as well?

**Comment by Nness** [comment]
@Andrew-Lahikainen 
Read only ticket make sense, but when come into query to find out which tickets he can see. That's a bit of tricky.

Current query is ticket must belong to same company as the user, ticket's board should be allowed, then based on all tickets, group tickets, site tickets to apply additional rule.

When cc kick in, in theory it should override any other rules. Since it is intentional. That means query need to add *include ticket that's not belong to user's company but has user's email in cc*. Here comes issue.
* Will any customer believe CC should still apply those permission rules? Such as board, group etc?
* If they don't like the rule we setup, will they want an option to turn it off?

Alternative approach will be switch account based on user's email, which we already have. Of course compare to above query change, this is not easy for user.

Switch account has similar affect as MSFT Teams guest account. So when people message you through teams, the mail need to clearly indicate which company's ticket. Then on click the link, user will logged into that company instead of his company. On top of that, he need to aware of the context.

Without switch context is always preferred. 

------------

Here is alternative proposal based on switch account. What if portal can facet tickets based on email and related accounts? e.g.
* My companies (6 open tickets)
* Ticket CC me (1 open tickets)
* Other companies (2 open tickets)

We allow him to check those list without switch account required. Also if ticket CC him and he doesn't have contact in given company, he can initialize guest account creation.

A lot of work to do on this, but overall experience feels better.

**Comment by nitesh305** [comment]
I think this alt would work well. We need this for our projects as it is important part of our business. I am also assuming many MSP's have this issue.
